### PR TITLE
Cfndsl206 apigateway resource

### LINKED ISSUE #206: Every Resource is of Type AWS::ApiGateway::Resource in version 0.9.1
Hey, Steven, 

This is a great tool! Thanks. 

I think a bug was introduced in version 0.9.1. When I run`cfndsl sample/autoscale.rb` all resources come out as `"Type": "AWS::ApiGateway::Resource"`. I maybe missing something, but I'm quite sure this is unexpected. Tested it out with version 0.9.0 and this behaviour does not exist.

I could not get my head around the problem, otherwise I'd gladly make a PR.

Hope this report helps. 

Thanks.

### PR DESCRIPTION
**Pull request by gergnz**
This fixes #206. Not sure if it's the best way, but it seems to work.

### CODE REVIEW

**Comment by stevenjack** [comment]
@gergnz thanks for looking at this, would you be able to stick a test in to check that we get back the correct resource type in the `Type` field? Another problem with the lacking set if tests we currently have that could have been easily picked up otherwise.

**Comment by gergnz** [comment]
I thought about it, but it was getting late.  I'll put something together in the next couple of days.

**Comment by gergnz** [comment]
@stevenjack. I think what I have done is appropriate and covers off at least Resources now.

File: spec/resources_spec.rb [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+      expect(subject.instance_variable_get('@Type')).to eq('AWS::EC2::Instance')
+    end
+  end
+end
 
+describe CfnDsl::ResourceDefinition do
+  subject { CfnDsl::CloudFormationTemplate.new.AutoScalingGroup(:web_servers) }
+  context '#all_refs' do
+    it 'checks that the type is AWS::AutoScaling::AutoScalingGroup' do
+      expect(subject.instance_variable_get('@Type')).to eq('AWS::AutoScaling::AutoScalingGroug')
```
Comment (kornypoet) [inline spec/resources_spec.rb]: Got a little typo here `Groug` instead of `Group`

**Comment by gergnz** [comment]
Thanks @kornypoet

File: spec/resources_spec.rb [hunk 2]
```diff
@@ -1,8 +1,21 @@
 require 'spec_helper'
 
 describe CfnDsl::ResourceDefinition do
-  subject { CfnDsl::CloudFormationTemplate.new.AutoScalingGroup(:web_servers) }
+  subject { CfnDsl::CloudFormationTemplate.new.EC2_Instance(:single_server) }
+  context '#all_refs' do
```
Comment (stevenjack) [inline spec/resources_spec.rb]: Not sure this context name is really very descriptive of what we're checking here, unless I'm missing something?

I'm going to merge this though as it's fixing a blocking problem and we can re-visit naming in the specs in a subsequent PR.

**Comment by stevenjack** [comment]
Apart from comment @gergnz this LGTM, thanks for adding the test in as well.

**Pull request merged**